components/display_drivers/include/gc9a01.hpp

```diff
@@ -332,62 +327,28 @@ class Gc9a01 {
     set_drawing_area(x, y, x + width, y + height);
 
     // Write the color data to controller RAM
-    send_command(Command::ramwr);
     uint32_t size = width * height;
     static constexpr int max_bytes_to_send = 1024 * 2;
     uint16_t color_data[max_bytes_to_send];
     memset(color_data, color, max_bytes_to_send * sizeof(uint16_t));
     for (int i = 0; i < size; i += max_bytes_to_send) {
-      int num_bytes = std::min((int)(size - i), (int)(max_bytes_to_send));
-      send_data((uint8_t *)color_data, num_bytes * 2);
+      int num_bytes = std::min(static_cast<int>(size - i), (int)(max_bytes_to_send));
+      write_command_(static_cast<uint8_t>(Command::ramwr), reinterpret_cast<uint8_t *>(color_data),
+                     num_bytes * 2, 0);
     }
   }
 
   /**
-   * @brief Sends the command, sets flags such that the pre-cb should set the
-   *        DC pin to command mode.
-   * @param command Command code to send
+   * @brief Send the provided commands to the display controller.
+   * @param commands Array of display_drivers::LcdInitCmd structures.
    */
-  static void send_command(uint8_t command) {
-    uint16_t flags = 0;
-    lcd_write_(&command, 1, flags);
-  }
-
-  /**
-   * @brief Sends the command, sets flags such that the pre-cb should set the
-   *        DC pin to command mode.
-   * @param command Command code to send
-   */
-  static void send_command(Command command) {
-    uint16_t flags = 0;
-    auto command_ = static_cast<uint8_t>(command);
-    lcd_write_(&command_, 1, flags);
-  }
-
-  /**
-   * @brief Send data to the display. Adds (1<<DC_LEVEL_BIT) to the flags so
-   *        that the pre-cb receives it in user data and can configure the DC
-   *        pin to data mode.
-   * @param data Pointer to array of bytes to be sent
-   * @param length Number of bytes of data to send.
-   * @param flags Optional (default = 0) flags associated with transfer.
-   */
-  static void send_data(const uint8_t *data, size_t length, uint32_t flags = 0) {
-    flags |= (1 << (int)display_drivers::Flags::DC_LEVEL_BIT);
-    lcd_write_(data, length, flags);
-  }
-
-  static void send_commands(display_drivers::LcdInitCmd *commands) {
+  static void send_commands(std::span<display_drivers::LcdInitCmd<>> commands) {
```

Comment:
might update this to have the arg be `const std::span<display_drivers::LcdInitCmd<> const>& commands` (or more concisely `const std::span<LcdInitCmd const> &commands` once/if you add the subclass + `using` directive.